src/main/resources/META-INF/resources/primefaces/colorpicker/1-colorpicker.js

```diff
@@ -192,13 +192,8 @@
      * @private
      */
     setupDialogSupport: function() {
-        var dialog = this.jqEl[0].closest('.ui-dialog');
-        if (dialog) {
-            var $dialog = $(dialog);
-
-            if($dialog.length == 1 && $dialog.css('position') === 'fixed') {
-                this.overlay.css('position', 'fixed');
-            }
+        if(!this.cfg.appendTo) {
```

Comment:
sure? so dialog support only works when appendTo is not configured by the user?

Reply:
No in the colorPicker case its only used in popup mode but Mert just refactored this code to bind the events on show and destroy them on hide. I am caching the original AppendTo value so we don't execute this code over and over every time the ColorPicker is show.  It is only needed once to correct the dialog CSS.

Reply:
You are right i have to recalculate the color picker every time setupDialogSupport is called.